packages/xo-server-backup-reports/src/index.js

```diff
@@ -182,7 +182,8 @@ class BackupReportsXoPlugin {
         // Handle improper value introduced by:
         // https://github.com/vatesfr/xen-orchestra/commit/753ee994f2948bbaca9d3161eaab82329a682773#diff-9c044ab8a42ed6576ea927a64c1ec3ebR105
         reportWhen === 'Never' ||
-        (reportWhen === 'failure' && log.status === 'success'))
+        (reportWhen === 'failure' && log.status === 'success') ||
+        (log.data.ignoreEmptyBackups && log?.result?.message === 'no VMs match this pattern'))
```

Comment:
We already discuss about this and I would prefer no testing on error messages.

I suggested you to simply not throw an error if the setting is enabled (maybe a warning or info though to avoid having an empty backup log).

Reply:
I agree that testing on error messages isn't great.

But the problem if we do what you suggest is that backup jobs that are configured with _Report when: always_ and _Don't send reports for backups with no matching VM: true_ will still send backup reports, even though the configuration tells that it shouldn't. Maybe we could just find another name to make it more clear, but I think this feature would make more sense if it doesn't send report in this case.

Also, it feels a bit strange to modify the result status of some backups (_Successfull_ instead of _Skipped_) just for the purpose of sending reports differently.

Reply:
I think a solution could be to add a boolean to the backup log whose value would indicate whether some VMs have been backed up or not. This would avoid the ugly string comparison in `xo-server-backup-reports`.

However, this boolean would be computed via error message comparison too, as this is the current way we detect *skipped* backups.

Reply:
What about implementing an option to not send reports when the run is skipped?

Reply:
It would be less convoluted solution. I think i like this idea

Reply:
Please check how easy it is to do and let me know :slightly_smiling_face: